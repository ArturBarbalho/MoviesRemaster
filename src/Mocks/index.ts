import { BannerProps, CardProps, bigbannerProps } from "../typesProps";
export const BiggerBannerMock:bigbannerProps = {
  bigsrc: 'https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg',
  smallsrc:'https://image.tmdb.org/t/p/original//ifRFLx83Xk1DcwAS3OScgI6HmWO.jpg',
  genres:['Action', 'Comedy'],
  id:2,
  media:'movie',
  overview:'Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...',
  title:'ONYANKOPON 3 O IN√çCIO'
}
export const CardMock:CardProps = {
  id:2,
  media:'movie',
  src:'http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg',
  title:'ONYANKOPON 2 O RETORNO'
}

export const BannerMock:BannerProps = {
    media: 'movie',
    relaseDate: '(1800)',
    id:1,
    title:'ONYANKOPON',
    src:'https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg',
    overview:'Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...',
    genres: ['Action', 'Comedy' ]
  }