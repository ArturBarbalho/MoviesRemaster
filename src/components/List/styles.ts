import styled from 'styled-components'
import media from 'styled-media-query'
export const Wrapper = styled.section`
display:flex;
flex-direction:row;
flex-wrap: wrap;
justify-content:center;
`