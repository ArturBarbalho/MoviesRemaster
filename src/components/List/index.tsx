import Card from '../Card'
import * as S from './styles'
import { CardProps } from '../../typesProps/index';
type props = {
  props : CardProps[]
}
const List = ({props}:props) => (
  <S.Wrapper>
    {props.map((e,i)=>{
      return <Card key={i} props={e} />
    })}
  </S.Wrapper>
)

export default List