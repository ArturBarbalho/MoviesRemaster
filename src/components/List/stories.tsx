import { Story, Meta } from '@storybook/react/types-6-0'
import List from '.'
import { CardMock } from '../../Mocks'

export default {
  title: 'List',
  component: List
} as Meta
const mock = [CardMock,CardMock,CardMock,CardMock,CardMock,CardMock,CardMock,CardMock]

export const Default: Story = () => <List props={mock} />