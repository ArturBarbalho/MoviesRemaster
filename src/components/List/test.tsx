import { render, screen } from '@testing-library/react'
import { CardMock } from '../../Mocks'

import List from '.'
const mock = [CardMock,CardMock]
describe('<List />', () => {
  it('should render the heading', () => {
    const { container } = render(<List props={mock} />)

    expect(screen.getAllByTestId('over')).toHaveLength(2)

    expect(container.firstChild).toMatchSnapshot()
  })
})