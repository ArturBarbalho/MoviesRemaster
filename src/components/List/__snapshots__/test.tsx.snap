// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should render the heading 1`] = `
<section
  class="styles__Wrapper-sc-g2zawo-0 gLOqk"
>
  <div
    class="styles__Wrapper-sc-1xbaftz-0 cSoiqW"
  >
    <div
      class="styles__Over-sc-1xbaftz-1 fRvyVk"
      data-testid="over"
    />
    <img
      class="styles__Wrapper-sc-y6e3xk-0 fOWEhk"
      src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
    />
  </div>
  <div
    class="styles__Wrapper-sc-1xbaftz-0 cSoiqW"
  >
    <div
      class="styles__Over-sc-1xbaftz-1 fRvyVk"
      data-testid="over"
    />
    <img
      class="styles__Wrapper-sc-y6e3xk-0 fOWEhk"
      src="http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg"
    />
  </div>
</section>
`;
